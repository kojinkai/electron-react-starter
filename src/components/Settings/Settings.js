import React, { Component } from 'react';

export class Settings extends Component {
  render() {
    return (
      <div className="settings">
        Rendering the settings
      </div>
    );
  }
}

export default Settings;
