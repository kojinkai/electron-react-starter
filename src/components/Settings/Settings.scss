@import 'styles/grid';

.settings {
  padding: $gutter;
  text-align: center;

  &__icon {
    display: block;
    margin: 3rem auto;
  }
}
