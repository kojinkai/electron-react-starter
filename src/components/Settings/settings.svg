<?xml version="1.0" encoding="UTF-8"?>
<svg width="114px" height="114px" viewBox="0 0 114 114" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 48.2 (47327) - http://www.bohemiancoding.com/sketch -->
    <title>Combined Shape</title>
    <desc>Created with Sketch.</desc>
    <defs></defs>
    <g id="Artboard-3" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd" transform="translate(-6.000000, -6.000000)">
        <path d="M93.1385648,112.23687 C88.0601779,115.352061 82.4628132,117.703603 76.5063231,119.131643 C75.8048817,117.267071 74.7039862,115.518533 73.2036366,114.018183 C67.5683237,108.38287 58.4316763,108.38287 52.7963634,114.018183 C51.2960138,115.518533 50.1951183,117.267071 49.4936769,119.131643 C43.5371868,117.703603 37.9398221,115.352061 32.8614352,112.23687 C33.6825796,110.423622 34.1397574,108.410383 34.1397574,106.290364 C34.1397574,98.320828 27.679172,91.8602426 19.7096361,91.8602426 C17.5896175,91.8602426 15.5763785,92.3174204 13.7631304,93.1385648 C10.6479394,88.0601779 8.2963974,82.4628132 6.86835708,76.5063231 C8.73292863,75.8048817 10.4814672,74.7039862 11.9818168,73.2036366 C17.6171297,67.5683237 17.6171297,58.4316763 11.9818168,52.7963634 C10.4814672,51.2960138 8.73292863,50.1951183 6.86835708,49.4936769 C8.2963974,43.5371868 10.6479394,37.9398221 13.7631304,32.8614352 C15.5763785,33.6825796 17.5896175,34.1397574 19.7096361,34.1397574 C27.679172,34.1397574 34.1397574,27.679172 34.1397574,19.7096361 C34.1397574,17.5896175 33.6825796,15.5763785 32.8614352,13.7631304 C37.9398221,10.6479394 43.5371868,8.2963974 49.4936769,6.86835708 C50.1951183,8.73292863 51.2960138,10.4814672 52.7963634,11.9818168 C58.4316763,17.6171297 67.5683237,17.6171297 73.2036366,11.9818168 C74.7039862,10.4814672 75.8048817,8.73292863 76.5063231,6.86835708 C82.4628132,8.2963974 88.0601779,10.6479394 93.1385648,13.7631304 C92.3174204,15.5763785 91.8602426,17.5896175 91.8602426,19.7096361 C91.8602426,27.679172 98.320828,34.1397574 106.290364,34.1397574 C108.410383,34.1397574 110.423622,33.6825796 112.23687,32.8614352 C115.352061,37.9398221 117.703603,43.5371868 119.131643,49.4936769 C117.267071,50.1951183 115.518533,51.2960138 114.018183,52.7963634 C108.38287,58.4316763 108.38287,67.5683237 114.018183,73.2036366 C115.518533,74.7039862 117.267071,75.8048817 119.131643,76.5063231 C117.703603,82.4628132 115.352061,88.0601779 112.23687,93.1385648 C110.423622,92.3174204 108.410383,91.8602426 106.290364,91.8602426 C98.320828,91.8602426 91.8602426,98.320828 91.8602426,106.290364 C91.8602426,108.410383 92.3174204,110.423622 93.1385648,112.23687 Z M63,93.0146523 C79.5766348,93.0146523 93.0146523,79.5766348 93.0146523,63 C93.0146523,46.4233652 79.5766348,32.9853477 63,32.9853477 C46.4233652,32.9853477 32.9853477,46.4233652 32.9853477,63 C32.9853477,79.5766348 46.4233652,93.0146523 63,93.0146523 Z" id="Combined-Shape" fill="#D8D8D8" transform="translate(63.000000, 63.000000) rotate(-315.000000) translate(-63.000000, -63.000000) "></path>
    </g>
</svg>