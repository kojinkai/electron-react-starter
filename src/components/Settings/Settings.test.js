import React from 'react';
import { shallow } from 'enzyme';
import Settings from './Settings';

it('renders according to the snapshot', () => {
  const wrapper = shallow(<Settings />);
  expect(wrapper).toMatchSnapshot();
});
