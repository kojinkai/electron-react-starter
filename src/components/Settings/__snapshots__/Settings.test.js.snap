// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders according to the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Settings />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="settings icon"
          className="settings__icon"
          src="settings.svg"
/>,
        "Add another page here",
      ],
      "className": "settings",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "settings icon",
          "className": "settings__icon",
          "src": "settings.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "Add another page here",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="settings icon"
            className="settings__icon"
            src="settings.svg"
/>,
          "Add another page here",
        ],
        "className": "settings",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "settings icon",
            "className": "settings__icon",
            "src": "settings.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "Add another page here",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
